class-two-factor-core.php

```diff
@@ -1804,34 +1839,35 @@ public static function manage_users_custom_column( $output, $column_name, $user_
 	public static function user_two_factor_options( $user ) {
 		$notices = [];
 
+		$providers = self::get_supported_providers_for_user( $user );
+
 		wp_enqueue_style( 'user-edit-2fa', plugins_url( 'user-edit.css', __FILE__ ), array(), TWO_FACTOR_VERSION );
 
 		$enabled_providers = array_keys( self::get_available_providers_for_user( $user ) );
-		$primary_provider_key  = self::get_primary_provider_key_selected_for_user( $user );
 
 		// This is specific to the current session, not the displayed user.
 		$show_2fa_options = self::current_user_can_update_two_factor_options();
 
-		if ( ! $show_2fa_options ) {
+		if ( $providers && ! $show_2fa_options ) {
 			$url = add_query_arg(
 				'redirect_to',
 				urlencode( self::get_user_settings_page_url( $user->ID ) . '#two-factor-options' ),
 				self::get_user_two_factor_revalidate_url()
 			);
 
 			$notices['warning two-factor-warning-revalidate-session'] = sprintf(
-					esc_html__( 'To update your Two-Factor options, you must first revalidate your session.', 'two-factor' ) .
+				esc_html__( 'To update your Two-Factor options, you must first revalidate your session.', 'two-factor' ) .
 					' <a class="button" href="%s">' . esc_html__( 'Revalidate now', 'two-factor' ) . '</a>',
-					esc_url( $url )
+				esc_url( $url )
 			);
 		}
 
-		printf(
-			'<fieldset id="two-factor-options" %s>',
-			$show_2fa_options ? '' : 'disabled="disabled"'
-		);
+		if ( empty( $providers ) ) {
+			$notices['notice two-factor-notice-no-providers-supported'] = esc_html__( 'No providers are available for your account.', 'two-factor' );
```

Comment:
Display a notice if no providers are available for user. 

I'm assuming that with the plugin enabled it would be more confusing to hide the section completely. With the section being present users can at least know the option should be available and is simply disabled.

Reply:
I agree, perhaps iterating on the copy to provide a user with some insight on what to do next?  Eg. "No providers are available for your account.  Please contact the site administrator for more details." or whatever verbage is used elsewhere in core to point folks to the admin (either site or network)?